### PR TITLE
[docs] Clarify restart strategy defaults set by checkpointing

### PR DESCRIPTION
**Pull request by rehevkor5**
- Added info about checkpointing changing the default restart
strategy in places where it was missing: the config page and the
section about the fixed-delay strategy
- Replaced no-restart with "no restart" so people don't think we're
 referring to a config value
- Replaced invalid <it> html tag with <code>
- Fixed bad link to restart strategies page from state.md

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by rehevkor5** [comment]
The changes so far are docs-only, so the CI failure must be unrelated.

File: docs/setup/config.md [hunk 1]
```diff
@@ -162,24 +162,27 @@ will be used under the directory specified by jobmanager.web.tmpdir.
...
-- `restart-strategy`: Default restart strategy to use in case that no restart strategy has been specified for the submitted job.
-Currently, it can be chosen from fixed delay restart strategy, failure rate restart strategy or no restart strategy.
-To use the fixed delay strategy you have to specify "fixed-delay".
-To use the failure rate strategy you have to specify "failure-rate".
-To turn the restart behaviour off you have to specify "none".
-Default value "none".
+- `restart-strategy`: Default [restart strategy]({{site.baseurl}}/dev/restart_strategies.html) to use in case no 
+restart strategy has been specified for the job.
+The options are:
+    - fixed delay strategy: "fixed-delay".
```
Comment (uce) [inline docs/setup/config.md]: Should we use code formatting with backticks instead of " for these as well?

File: docs/dev/restart_strategies.md [hunk 2]
```diff
@@ -28,8 +28,9 @@ In case that the job is submitted with a restart strategy, this strategy overrid
 
 The default restart strategy is set via Flink's configuration file `flink-conf.yaml`.
 The configuration parameter *restart-strategy* defines which strategy is taken.
-Per default, the no-restart strategy is used.
-When checkpointing is activated and no restart strategy has been configured, the job will be restarted infinitely often.
+If checkpointing is not enabled, the "no restart" strategy is used.
+If checkpointing is activated and the restart strategy has not been configured, the fixed-delay strategy is used with
```
Comment (uce) [inline docs/dev/restart_strategies.md]: Should we say "If checkpointing is activated and no restart strategy has been explicitly configured, ..."

**Comment by uce** [comment]
Thanks for these clarifications. I'm going to merge it with the next batch. In the current master we also have a new checkpoints page at `setup/checkpoints.html` where we can add these notes as well.

**Comment by uce** [comment]
Thanks again. Just merged this to `master` (b03ad79) and `release-1.2` (2aaa093).

**Comment by rehevkor5** [comment]
@uce Thanks. I didn't get an opportunity to address your comments... is there anything you want me to do?

**Comment by rehevkor5** [comment]
I'm curious as to how you added my commit to `master` and `release-1.2` but it doesn't show up in this PR. The whole "rehevkor5 committed with uce 5 days ago" thing. Github isn't aware that the changes in this PR were merged.

If you have a moment can you describe your workflow is so I can understand what's happening behind the scenes? Is it rebase onto master + merge --ff-only or something? Maybe it's due to the fact that Github is only mirroring another repo? I'd appreciate it!

**Comment by uce** [comment]
Since the comments were very minor I addressed them myself (used the backticks and skipped the change in "If checkpointing is activated..." as it didn't really improve anything).

The work flow was as follows: I checked out your PR branch, rebased it onto master (via `git rebase apache/master`) and pushed it with another related commit. My related commit had a comment `This closes #3059`, which tells the ASF bot to close this PR. Then I cherry picked it to the 1.2 branch.

GitHub only notices that the changes were merged for merges to master and if the commit hashes are not changed I think. Since I rebased they did change and I needed the manual close. Does this make sense?

**Comment by rehevkor5** [comment]
It does, thanks! I'll try to remember to include "closes #_{pr number}_" in my commit messages which might be handy.

**Pull request closed without merging**